building/build-debs/homeworld-admin-tools/src/access.py

```diff
@@ -52,15 +64,32 @@ def call_keyreq(keyreq_command, *params, collect=False):
     with tempfile.TemporaryDirectory() as tdir:
         https_cert_path = os.path.join(tdir, "server.pem")
         util.writefile(https_cert_path, authority.get_pubkey_by_filename("./server.pem"))
-        try:
-            return invoke_variant(["keyreq", keyreq_command, https_cert_path, keyserver_domain] + list(params))
-        except subprocess.CalledProcessError as e:
-            if e.returncode == 1:
-                fail_hint = "do you have valid kerberos tickets?\n" \
-                    "or, your connection to the server might be faulty.\n" \
-                    "or, the server's keygateway might be broken."
-                command.fail("keyreq failed: %s" % e, fail_hint)
-            raise e
+        keyreq_sp = subprocess.Popen(["keyreq", keyreq_command, https_cert_path, keyserver_domain] + list(params), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        output, err_bytes = keyreq_sp.communicate()
+        if keyreq_sp.returncode != 0:
+            error_code_meaning = "unknown error code"
+            if keyreq_sp.returncode in KEYREQ_ERROR_CODES:
+                error_code_meaning = KEYREQ_ERROR_CODES[keyreq_sp.returncode]
+            err = err_bytes.decode()
+            print(err)
+            fail_hint = None
+            if keyreq_sp.returncode == 2:
+                knc_stderr_start = err.find(KNC_STDERR_START_TAG)
+                knc_stderr_end = err.find(KNC_STDERR_END_TAG)
+                if knc_stderr_start != -1 and knc_stderr_end != -1:
+                    knc_stderr = err[knc_stderr_start + len(KNC_STDERR_START_TAG):knc_stderr_end]
+                    if "gstd_initiate: continuation failed" in knc_stderr:
+                        fail_hint = "the server's keygateway might be broken."
+                    elif "gss_init_sec_context: No Kerberos credentials available" in knc_stderr or "gstd_error: gss_init_sec_context: Ticket expired" in knc_stderr:
+                        fail_hint = "do you have valid kerberos tickets?"
+                elif "empty response, likely because the server does not recognize your Kerberos identity" in err:
+                    fail_hint = "your kerberos tickets might be for the wrong instance."
+            fail_message = "keyreq failed with error code %d: %s" % (keyreq_sp.returncode, error_code_meaning)
+            if fail_hint is None:
+                command.fail(fail_message)
+            else:
+                command.fail(fail_message, fail_hint)
```

Comment:
This entire block of added code is fairly messy, and hard to decipher. Can you try to find some way to clean it up and make it more concise?